### PR TITLE
Implemented new virtual disk streaming I/O.

### PR DESCRIPTION
**Pull request by roliveri**
Implemented module-based disk IO streaming chain.

### CODE REVIEW

**Comment by movitto** [comment]
@roliveri I went thought this and the code looks good. I didn't verify the DOS / MBR / GPT standards but successfully ran the spec suite (which are the testing the attached disks which have assumably been formatted with the corresponding data structures?) The was one small change I had to make to get the tests running locally, will comment inline (may just be due to an older dep on my system).

The FileIO class is intended to replace the current 'BlockFile' class correct? If so we can switch over the existing virtfs modules to use that instead and remove that from here. In any case, everything else looks good, have no problem w/ merging this

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  require "logger"
+
+  log = Logger.new(STDERR)
+  log.formatter = lambda { |_severity, _datetime, _progname, msg| "#{msg}\n" }
+  log.level = case ENV['LOG_LEVEL']
+              when 'ERROR' then Logger::ERROR
+              when 'WARN'  then Logger::WARN
+              when 'INFO'  then Logger::INFO
+              when 'DEBUG' then Logger::DEBUG
+              else              Logger::OFF
```
Comment (movitto) [inline spec/spec_helper.rb]: I'm getting an "undefined constant Logger::OFF" here, changing it to one of the other values is a workaround

Comment (roliveri) [reply spec/spec_helper.rb]: @movitto I converted this from Log4r to Logger, to eliminate the Log4r dependancy. I guess Logger doesn't have an "OFF" level. I'll change the default to "ERROR".

**Comment by roliveri** [comment]
@movitto `FileIO` replaces `BlockFile` for regular files, which is what we need for most of the cases. We'll probably need a different module to access block special files, like `/dev/sda`.

**Comment by roliveri** [comment]
@movitto Now that this is merged, try converting some of your existing VirtFS FS modules to use this.

**Pull request merged**